app/services/generate_application_ref.rb

```diff
@@ -0,0 +1,16 @@
+class GenerateApplicationRef
+  REFERENCE_LENGTH = 6
+
+  class << self
+    def call(application_form)
+      application_form.update_attribute(:reference, new_application_reference)
+    end
+
+    def new_application_reference
+      letters = ('A'..'Z').to_a
+      digits = ('0'..'9').to_a
```

Comment:
We need to omit easily confused characters? For example `0O1I`

Reply:
There's also the risk that this generates rude words.

Reply:
We also need to ensure this is distinct form any UCAS reference numbers.

Reply:
Numbers are easier to read out over the phone.

This is a related issue in the design system:
https://github.com/alphagov/govuk-design-system-backlog/issues/85

Reply:
Based on this feedback, I propose:

- Don't use 0 and 1 in `digits`
- Don't use `I` (for India), `L` (for Lima) in `letters` (`letters - ['I', 'L']`)
- Use a reference format that combines 2 letters, 4 numbers e.g. `XX3333`

Reply:
@tvararu implementation sounds good, I'm working on it.
Also, I've just realised that we need to check that the code is not already used (in the remote chance it was previously generated)

Reply:
@fofr @paulrobertlloyd here are 100 samples generated useing the logic suggested by @tvararu:

```"TN6267 PG3136 EQ2186 NX5371 ZA8388 WY2929 KT8181 BX3494 AN6562 CV2756 ZG1569 MK6699 PZ1581 JV1456 AA9346 FE1671 BE8362 RZ7144 TQ1273 PB4545 CA3734 PW4148 UR4118 WN3213 GB8262 VR7787 VG8294 XR9933 WF6226 BZ4141 HD1393 YV6311 YJ6751 QB9197 JY7786 NH3725 PV6993 AD9128 XW7454 SA4593 GZ2328 CH1439 YG2945 AH3891 MT2653 HK6684 EZ8486 JV6887 TS6662 KR3135 UZ1212 UM3618 KT7753 YU5433 UC7163 TJ2879 FX4716 TA6311 JZ9654 YQ4395 GU6442 DE7263 ZN6388 VW2986 AC3668 AM6129 MS8643 BF9838 KY9421 CV1195 VM9365 FP6824 PR1681 XV9739 HJ9883 MT4832 DF7634 UJ9874 KR4898 YX7368 AK6315 JC5391 DS7832 MB1182 KG2255 WJ3588 YY3629 AZ9144 VD9466 YJ3777 CW3387 HK5935 YF5789 VX1319 GA9483 YA1547 KH1725 UQ5434 WP1283 UB2713```